src/hooks/__internal__/useCharacterCount/useCharacterCount.tsx

```diff
@@ -1,22 +1,29 @@
-import React, { useMemo, useRef } from "react";
+import React, { useMemo, useRef, useEffect, useState } from "react";
 import CharacterCount from "../../../__internal__/character-count";
-import useLocale from "../useLocale";
 import guid from "../../../__internal__/utils/helpers/guid";
+import useDebounce from "../useDebounce";
 
 const useCharacterCount = (
   value = "",
   characterLimit?: number,
   enforceCharacterLimit = true
-): [
-  number | undefined,
-  JSX.Element | null,
-  string | undefined,
-  string | null
-] => {
+): [number | undefined, JSX.Element | null, string | undefined] => {
   const isCharacterLimitValid =
     typeof characterLimit === "number" && !Number.isNaN(characterLimit);
-  const l = useLocale();
-  const hintString = l.characterCount.hintString();
+
+  const [debouncedValue, setDebouncedValue] = useState(value);
```

Comment:
**suggestion(non-blocking):** Not going to make you do this now as I know we have time constraints on this. It might be worth considering in the future to potentially use a ref here instead of `useState`. For now the performance benefits will be negligible though. I'd rather the work be done and we can focus on incremental improves in the future 😃